src/storage/src/storage_state.rs

```diff
@@ -1072,43 +1085,38 @@ impl<'w, A: Allocate> Worker<'w, A> {
                             let running_ingestion =
                                 self.storage_state.ingestions.get(&ingestion.id);
 
-                            // Ingestion statements are only considered updates if they are
-                            // currently running.
-                            ingestion.update = running_ingestion.is_some();
-
                             // We keep only:
                             // - The most recent version of the ingestion, which
                             //   is why these commands are run in reverse.
                             // - Ingestions whose descriptions are not exactly
                             //   those that are currently running.
-                            seen_most_recent_ingestion.insert(ingestion.id)
+                            seen_most_recent_definition.insert(ingestion.id)
                                 && running_ingestion != Some(&ingestion.description)
```

Comment:
i wonder if we actually need both of these checks... i think the first should cover everything right?

Reply:
That's probably right––I have it on my list of things to go through reconciliation and tidy things up. I skipped over fixing this just for my own sanity in trying to get this merged.